worlds/tunic/__init__.py

```diff
@@ -412,12 +429,34 @@ def fill_slot_data(self) -> Dict[str, Any]:
             "disable_local_spoiler": int(self.settings.disable_local_spoiler or self.multiworld.is_race),
         }
 
+        # this would be in a stage if there was an appropriate stage for it
+        self.player_item_link_locations = {}
+        groups = self.multiworld.get_player_groups(self.player)
+        if groups:
+            if not self.item_link_locations:
+                tunic_worlds: Tuple[TunicWorld] = self.multiworld.get_game_worlds("TUNIC")
+                # figure out our groups and the items in them
+                for tunic in tunic_worlds:
+                    for group in self.multiworld.get_player_groups(tunic.player):
+                        self.item_link_locations.setdefault(group, {})
+                for location in self.multiworld.get_locations():
+                    if location.item and location.item.player in self.item_link_locations.keys():
+                        (self.item_link_locations[location.item.player].setdefault(location.item.name, [])
+                         .append((location.player, location.name)))
```

Comment:
I don't really like the fix here because I think that if there are item links, but **no Tunic world is participating in the itemlinks**, every Tunic world will iterate the whole itempool

Reply:
If that is true, I would suggest just having item_links_locations be None to start with and to change this to `if self.item_link_locations is None`

Reply:
At least while groups are only used for item links `groups = self.multiworld.get_player_groups(self.player)` is only a truthy value if the current TUNIC player is participating in item links.

Then, if `self.item_link_locations` is empty, an empty dict `{}` is set into `self.item_link_locations` for each `group` that any TUNIC player belongs to, so `self.item_link_locations` will be truthy from that point onwards because it is known at that point that at least the current TUNIC player is participating in item links.

I suggested having `item_links_locations` starting off as `None` and checking that first instead of checking if the current TUNIC player belongs to a group, but Scipio decided against it. https://github.com/ArchipelagoMW/Archipelago/pull/4105#discussion_r1818152370

Reply:
You're right. Nowadays, I'm more and more moving to the position that code that's like, "If you think it through with every scenario, it works", is pretty bad. But it does work, so I guess it's fine :/